packages/examples/usage-tracking/usage-tracking.js

```diff
@@ -0,0 +1,106 @@
+  numConversations = data.length;
+
+  console.log(`Got back ${data.length} conversations`);
+  data.forEach((element) => {
+    console.log('------------------------------------------------------------\n');
+    console.log(`Conversation ID: ${element.id}`);
+    console.log(`Total conversation turns\t${element.turns.length}`);
+
+    getConvoTurns(element);
+
+    // We want to get content for any messages that are:
+    // conversation.role = "user" -> get messages that are messages.kind = "text" and get messages.content for the text of the message
+    //      make sure the state is "done"
+    // conversation.role = "assistant"
+    //      kind = "text", get the content
+    //      make sure the state is "done"
+    //
+    //      kind = "functionResponse"....get the response
+  });
+
+  console.log('\n\n============================================================');
+  console.log(`Final stats for agent ${AGENT_ID}:\n`);
+  console.log(`Total Conversations\t${numConversations}\n`);
+  console.log(`Total Agent Messages\t${numMessages}\n`);
+  console.log(`Total Characters\t${lenConversations}\n`);
+  console.log(`Total LLM Tokens\t${numTokens}\n`);
+  console.log('============================================================');
+});
+
+function getConvoTurns(conversation) {
+  let numConvoMessages = 0;
+  let numConvoChars = 0;
+  let numConvoTokens = 0;
+
+  // Iterate thru the conversation and process all the turns
+  conversation.turns.forEach((turn) => {
+    numConvoMessages += turn.messages.length;
+
+    // Iterate through the turn messages and log their stats
+    turn.messages.forEach((message) => {
+      // console.log(message);
+      numMessages++;
+      // Make sure it's a message type that we want
+      if (turn.state == 'done') {
+        // Function Response
+        if (turn.role == 'assistant' && message.kind == 'functionResponse') {
+          lenConversations += message.response.length;
+          const encoded = encode(message.response);
+          numTokens += encoded.length;
+
+          numConvoChars += message.response.length;
+          numConvoTokens += encoded.length;
+        } else if (message.kind == 'text' && (turn.role == 'assistant' || turn.role == 'user')) {
+          lenConversations += message.content.length;
+          const encoded = encode(message.content);
+          numTokens += encoded.length;
+
+          numConvoChars += message.content.length;
+          numConvoTokens += encoded.length;
+        }
+      }
+    });
+  });
+
+  console.log(`Total conversation messages\t${numConvoMessages}`);
+  console.log(`Total conversation characters\t${numConvoChars}`);
+  console.log(`Total conversation tokens\t${numConvoTokens}`);
+}
+
+async function getAgentConversations(agentId) {
+  try {
+    const response = await axios({
+      method: 'get',
+      maxBodyLength: Infinity,
+      url: `https://api.fixie.ai/api/v1/agents/${agentId}/conversations`,
```

Comment:
Is this endpoint paginated?

Reply:
yes it is. changed the implementation to handle paging.